sdk/identity/azure-identity/src/token_credential_impl.cpp

```diff
@@ -448,13 +449,21 @@ AccessToken TokenCredentialImpl::ParseToken(
           try
           {
             accessToken.ExpiresOn = parse(expiresOnAsString);
-            return accessToken;
+            // Workaround for Warning C26800 - Use of a moved from object: 'accessToken'
+            // (lifetime.1) on MSVC version 14.40.33807+.
+            // Returning accessToken here directly causes the warning.
+            successfulParse = true;
+            break;
```

Comment:
I think we might have got this after the nlohmann json update.
Instead of all this, if you change `auto const` on line 424 to be `std::string const`, the warning should go away.

Reply:
That didn't address the warning. The alternative is for me to suppress this warning on Windows, since it seems like an MSVC bug. That might be better than contorting the code.

Reply:
Hmm, looks like a faulty warning then, but I think it is safer to not suppress it then - what if it actually does use the moved value like the warning says, then we are in UB territory.

Reply:
If you suspect that the warning is faulty, start a thread on vcsig - the folks who own the C++ compiler can chime in to verify.

It will help if you can come up with a small example in compiler explorer, FWIW.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/5741